.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,27 @@
+The following fields can give you some guidance on formatting your
+pull request. Not all fields are relevant for all pulls. Please
+fill in fields that you think are useful; brevity and clarity are
+valued.
+
+-----
+
+### About
+
+Briefly describe what you did, if you need to include technical details, start here.
+
+### Issues
+
+Fixes #012
+
+### Reason
+
+If this change doesn't fix an issue, it would be good to give some background as to why this change is necessary.
+
+### Release Notes
+
+Include possible release notes/documentation, if applicable.
```

Comment:
I think maintaining a CHANGELOG might be more trouble than it's worth; because either:
- updating it is part of the pull request, and it's easy to get conflicts with multiple concurrent pull requests are active
- or it must be updated manually during or after the merge

It's probably easier to just create release notes when about to release from tracker activity, assuming we don't wait 2 years between releases :P.